docs/keps/20200619-federated-resource-status.md

```diff
@@ -50,14 +51,38 @@ resources are healthy or not across clusters.
 
 * Quickly identify unhealthy federated resources by relying on the status of the federated resources.
 * Improve the troubleshooting of failures when propagating resources.
+* Define a new api version potentially `v1beta2` to include these new fields.
 
+### Non Goals
+
+* Set the resource status for all the federated resources and clusters whenever a federated resource satisfies the condition `Ready=True`.
+This goal would imply adding new flags to the kubefed control-plane components to limit the value and the number of concurrent requests that can be processed by the API server. So the control-plane is not blown by the amount of requests based on the metric of existing federated resources and kubefed clusters.
```

Comment:
This is quite unclear. Do you mean usage of a flag as a controller config? Or a flag on the federated type config. Also how would using a flag solve the problem of not blowing up the control-plane? I mean just the flag/switch does not solve the amount of requests. Some adaptive mechanism, which can reduce the amount of requests can only do that.

Reply:
@irfanurrehman Yes, the approach includes a `flag` in the control plane, if you wanna know more about this kind of flag, it'd suggest to check the `kube-api-qps` or `kube-api-burst` flag used by the controller-manager, kube-proxy, ..., to talk to the Kubernetes API Server. I didn't want to throw too many details and leave it a bit open (we can always update this KEP). 



More details:

- Push reconciler: the kubefed controller plane talks to the Kubernetes API server of the host cluster, as well as the e Kubernetes API servers of the kubefed clusters. Therefore I considered we should define this flag (qps,burst), thus when creating the clients to the remote Kubernetes API server we can limit the amount of requests per seconds etc. This becomes more important when enabling the collection of the status from the resources when the amount of requests would increase.



- Pull reconciler: In this scenario, the flag values would be used to talk to the kubefed shim API server from the kubefed daemons, but it follows the same logic. The client connection would use these values to create the Kubernetes client and limit the clients.

Reply:
@irfanurrehman wdyt?

Reply:
Suggestion: s/This goal would imply adding new flags to the kubefed control-plane components to limit the value and the number of concurrent requests that can be processed by the API server. So the control-plane is not blown by the amount of requests based on the metric of existing federated resources and kubefed clusters./Addition of feature flags to control plane components to limit or throttle requests made to kubernetes API servers akin to  `kube-api-qps` or `kube-api-burst` in kubelet./